core/src/main/java/org/infinispan/marshall/exts/DoubleSummaryStatisticsExternalizer.java

```diff
@@ -56,33 +78,51 @@ private void verifySerialization() {
    @Override
    public void writeObject(ObjectOutput output, DoubleSummaryStatistics object) throws IOException {
       verifySerialization();
-      try {
-         output.writeLong(countField.getLong(object));
-         output.writeDouble(sumField.getDouble(object));
-         output.writeDouble(sumCompensationField.getDouble(object));
-         output.writeDouble(simpleSumField.getDouble(object));
-         output.writeDouble(minField.getDouble(object));
-         output.writeDouble(maxField.getDouble(object));
-      } catch (IllegalAccessException e) {
-         // This can't happen as we force accessibility in the getField
-         throw new IOException(e);
-      }
+      output.writeLong(object.getCount());
+      output.writeDouble(object.getSum());
+      // To stay backward compatible with older versions of infinispan, write 0s for the compensation and simpleSum
+      // fields. The API doesn't allow for obtaining nor setting values of these _implementation specific_ fields.
+      // This means precision can be lost.
+      output.writeDouble(0); // sumCompensation
+      output.writeDouble(0); // simpleSum
+      output.writeDouble(object.getMin());
+      output.writeDouble(object.getMax());
    }
 
    @Override
    public DoubleSummaryStatistics readObject(ObjectInput input) throws IOException, ClassNotFoundException {
       verifySerialization();
-      DoubleSummaryStatistics summaryStatistics = new DoubleSummaryStatistics();
-      try {
-         countField.setLong(summaryStatistics, input.readLong());
-         sumField.setDouble(summaryStatistics, input.readDouble());
-         sumCompensationField.setDouble(summaryStatistics, input.readDouble());
-         simpleSumField.setDouble(summaryStatistics, input.readDouble());
-         minField.setDouble(summaryStatistics, input.readDouble());
-         maxField.setDouble(summaryStatistics, input.readDouble());
-      } catch (IllegalAccessException e) {
-         // This can't happen as we force accessibility in the getField
-         throw new IOException(e);
+      final DoubleSummaryStatistics summaryStatistics;
+      if (constructor != null) {
+         // JDK 10+, pass values to constructor
+
+         final long count = input.readLong();
+         final double sum = input.readDouble();
+         final double sumCompensation = input.readDouble(); // ignored, see writeObject()
+         final double simpleSum = input.readDouble(); // ignored, see writeObject()
+         final double min = input.readDouble();
+         final double max = input.readDouble();
```

Comment:
I'd move the stream reads before the `if`, so it's clearer that the format is the same regardless of JDK version.

Reply:
Good point.